audinterface/core/feature.py

```diff
@@ -415,6 +417,92 @@ def to_numpy(
         """
         return frame.values.T.reshape(self.num_channels, self.num_features, -1)
 
+    def _assert_shape_3d(
+            self,
+            features: typing.Union[np.ndarray, pd.Series]
+    ):
+        r"""Reshape to [n_channels, n_features, n_frames]."""
+
+        if self.process.process_func_is_mono:
+            features = np.array(features)
```

Comment:
If we don't have `self.process.process_func_is_mono` we raise an error if `features` is not a numpy array.
Why is it ok here if we don't have one?

Reply:
The process function still has to return `np.array`, but internally we loop over all channels and collect the results in a list and here we convert that list to a `np.array`, i.e. we combine the results of the channels to a single array.

Reply:
But will it still raise an error in this case if the process function does not return `np.array` and we have `process_func_is_mono`?

Reply:
Oh now I get your point. Yes, I think you are right, we actually need to assert that every element of the list is a `np.ndarray`.

Reply:
I fixed it and added a test for it.

Reply:
Great, looks good now.